x-pack/solutions/observability/plugins/observability_ai_assistant_app/server/functions/documentation.ts

```diff
@@ -69,7 +69,7 @@ export async function registerDocumentationFunction({
         max: 3,
         connectorId,
         request: resources.request,
-        functionCalling: useSimulatedFunctionCalling ? 'simulated' : 'native',
+        functionCalling: useSimulatedFunctionCalling ? 'simulated' : 'auto',
```

Comment:
As mentioned above, I'd pass `functionCallingMode` through instead of `useSimulatedFunctionCalling`

```suggestion
        functionCalling: functionCallingMode,
```